Code/DistGeom/DistViolationContribs.cpp

```diff
@@ -64,16 +66,16 @@ void DistViolationContribs::getGrad(double *pos, double *grad) const {
   const unsigned int dim = this->dp_forceField->dimension();
 
   auto contrib = [&](const auto &c) {
-    double d = distance(c.idx1, c.idx2, pos, dp_forceField->dimension());
+    double d2 = distance2(c.idx1, c.idx2, pos, dp_forceField->dimension());
+    double d;
     double preFactor = 0.0;
-    if (d > c.ub) {
-      double u2 = c.ub * c.ub;
-      preFactor = 4. * (((d * d) / u2) - 1.0) * (d / u2);
-    } else if (d < c.lb) {
-      double d2 = d * d;
-      double l2 = c.lb * c.lb;
-      double l2d2 = d2 + l2;
-      preFactor = 8. * l2 * d * (1. - 2 * l2 / l2d2) / (l2d2 * l2d2);
+    if (d2 > c.ub2) {
+      d = sqrt(d2);
+      preFactor = 4. * (((d2) / c.ub2) - 1.0) * (d / c.ub2);
```

Comment:
Based on what I saw on linux, I think that the CI builds will go through if you replace this with:
```suggestion
      preFactor = 4. * (((d * d) / c.ub2) - 1.0) * (d / c.ub2);
```

Reply:
Hi Greg,

Thank you for your help with the investigation. I noticed that my changes were failing the tests on Mac and left this PR as a draft to further investigate the issues. I appreciate your suggestions and will add the changes as you proposed. If the changes resolve the issues and the tests pass successfully, I will update the PR and make it ready for review.